src/styles/_calendar.scss

```diff
@@ -0,0 +1,54 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: 10px;
+  width: calc((#{$day-size} * #{$columns}) + (#{$gap} * #{$columns - 1}));
+  background-color: $calendar-background-color;
+
+  &__day {
+    box-sizing: border-box;
+    display: flex;
+    border: 1px solid $border-color;
+    width: $day-size;
+    height: $day-size;
+    background-color: $day-background-color;
+    transition:
+      background-color 0.5s,
+      transform 0.5s;
+    &::before {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      margin: auto;
+    }
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-background-color;
+      transform: translateY(-20px);
+      transition:
+        background-color 0.5s,
+        transform 0.5s;
+    }
+  }
+
+  @each $day in $days {
+    $index: list.index($days, $day) - 1;
+    &--start-day-#{$day} .calendar__day:nth-child(1) {
+      margin-left: $index * ($day-size + 1px);
```

Comment:
The calculation for margin-left might be incorrect if the intention is to space the first day according to the start day of the week. Ensure that the calculation considers the correct unit and value for spacing.